BookingPages/CarRentalsPage.cs

```diff
@@ -1,19 +1,21 @@
 ﻿using OpenQA.Selenium;
+using System.Collections.ObjectModel;
+using Utilities;
+using Wrappers;
 
 namespace BookingPages
 {
     public class CarRentalsPage
     {
         private readonly IWebDriver _driver;
-        private IWebElement SearchButton => _driver.FindElement(By.CssSelector(".submit-button-container"));
-        private IWebElement ErrorMessage => _driver.FindElement(By.XPath("//*[text()='Please provide a pick-up location']"));
-        private IWebElement PickUpLocation => _driver.FindElement(By.CssSelector("#searchbox-toolbox-fts-pickup"));
-        private IWebElement PickUpDate => _driver.FindElement(By.XPath("//*[@data-testid ='searchbox-toolbox-date-picker-pickup-date-value']"));
-        private IWebElement DropOffDate => _driver.FindElement(By.XPath("//*[@data-testid ='searchbox-toolbox-date-picker-dropoff-date-value']"));
-        private IWebElement Calendar => _driver.FindElement(By.XPath("//*[@data-testid='bui-calendar']"));
-        private IWebElement NextMonthArrow => Calendar.FindElement(By.XPath("//*[@data-testid='bui-calendar']/ button"));
-        private IList<IWebElement> CurrentMonths => Calendar.FindElements(By.XPath("//*[@data-testid='bui-calendar']//h3"));
-        private IList<IWebElement> CurrentDays => Calendar.FindElements(By.XPath("//*[@data-testid='bui-calendar']//td"));
+        private By CurrentDays => (By.XPath("//*[@data-testid='bui-calendar']//td"));
+        private Button SearchButton => new Button(_driver.FindElement(By.CssSelector(".submit-button-container")));
+        private WebPageElement ErrorMessage => new WebPageElement(_driver.FindElement(By.XPath("//*[text()='Please provide a pick-up location']")));
+        private TextBox PickUpLocation => new TextBox(_driver.FindElement(By.CssSelector("#searchbox-toolbox-fts-pickup")));
```

Comment:
use constructor that accepts only locator, no need in driver in pages classes

Reply:
ok, changed everywhere